lib/data/submission.js

```diff
@@ -322,5 +322,31 @@ const diffSubmissions = (structurals, versions) => {
   return byVersion;
 };
 
-module.exports = { submissionXmlToFieldStream, getSelectMultipleResponses, _hashedTree, _diffObj, _diffArray, diffSubmissions, _symbols: { subhash, subhashes, keys, score } };
+////////////////////////////
+// ENTITY PARSING
+//
+// Parse submission XML and use with form field -> entity property mappings to build entities.
+
+// n.b. this wants only entity fields with filled in entity_prop attribute!
+const getEntity = (entityFields, xml) => new Promise((resolve, reject) => {
+  const values = {};
+  const stream = submissionXmlToFieldStream(entityFields, xml);
+  stream.on('error', reject);
+  stream.on('data', ({ field, text }) => {
+    if (field.path === '/meta/entity/label')
```

Comment:
should look for entity label and entity create. these will be form fields so there's probably another way to get those fields and pass them in.